src/target/adiv5_swd.c

```diff
@@ -79,133 +135,200 @@ static uint32_t firmware_dp_low_read(const uint16_t addr)
 	return res == SWDP_ACK_OK ? data : 0;
 }
 
-/* Try first the dormant to SWD procedure.
- * If target id given, scan DPs 0 .. 15 on that device and return.
- * Otherwise
- */
-uint32_t adiv5_swdp_scan(uint32_t targetid)
+uint32_t adiv5_swdp_scan(const uint32_t targetid)
```

Comment:
Given this appears to return 0 for failure and 1 for success, does it make sense to change this to return `bool` and clean up the returns accordingly?

If we do that, lets also rename this `adiv5_swd_scan()` as the 'p' in the name is unhelpful - it's scanning the SWD bus. That puts the nomenclature in line with the JTAG scan engine

Reply:
It does make sense, from what I gathered reading the code, I did it, but the changes started to cascade a bit so I think I'll submit that later, in an effort to get this merged faster

With that said I took the name note and renamed multidrop scan to align with the clearer nomenclature

Reply:
Ah, sure. We can address that in a follow-up cleanup PR then. You're right that it probably does have some far reaching consequences and so will cause a lot of churn.